api/provider/aws/retry.go

```diff
@@ -0,0 +1,24 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func Retry(id string, describeFN func(id string) error) error {
+	timeout := time.After(config.DefaultRetryTimeOut)
+	tick := time.Tick(config.DefaultRetryWaitTime)
+
+	for {
+		select {
+		case <-timeout:
```

Comment:
[nit] can just use `<-time.After(timeout)` here. Also, in the examples I've seen for golang timeouts, the timeout case is at the bottom of the select statement - similar to how the `default` is usually at the bottom as well.